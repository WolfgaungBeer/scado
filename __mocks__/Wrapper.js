import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { Theme, defaultTheme } from '../scado';

const mockStore = configureStore();
const store = mockStore({
    scado: {
        theme: defaultTheme,
        toasts: [
            { id: '1', type: 'success', content: 'Text', onClose: () => {} },
            { id: '2', type: 'error', content: 'Text', onClose: () => {} },
        ],
    },
});


const Wrapper = ({ children }) => ( // eslint-disable-line
    <Provider store={store}>
        <Theme>{children}</Theme>
    </Provider>
);

export default Wrapper;
