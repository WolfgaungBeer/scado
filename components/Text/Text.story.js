import React from 'react';
import { storiesOf } from '@storybook/react';
import { text } from '@storybook/addon-knobs';
import Text from './';
import Icon from '../Icon';


storiesOf('Text', module)
    .add('default', () => (
        <Text>Das ist ein Text</Text>
    ))
    .add('with props', () => (
        <Text
            color={text('color', 'success')}
            scale={text('scale', 'l')}
        >Das ist ein Text</Text>
    ))
    .add('with an icon on the left side', () => (
        <div>
            <Icon icon="dashboard" color="success" scale="xl" />
            <Text color="success" scale="xl">Das ist ein Text</Text>
        </div>
    ))
    .add('with an icon on the right side', () => (
        <div>
            <Text color="success" scale="xl">Das ist ein Text</Text>
            <Icon icon="dashboard" color="success" scale="xl" />
        </div>
    ))
    .add('with an nested icon', () => (
        <Text color="success" scale="xl">
            <Icon icon="dashboard" color="success" scale="xl" />
            Das ist ein Text
        </Text>
    ));
