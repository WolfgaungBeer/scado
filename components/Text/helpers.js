export const textStyle = () => props => `
    color: ${props.theme.color[props.color]};
    font-size: ${props.theme.scale[props.scale]};
    font-family: ${props.theme.fonts.paragraph};
`;

export const font = type => props => `
    font-family: ${props.theme.fonts[type]};
`;
