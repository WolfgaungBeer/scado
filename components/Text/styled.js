import styled from 'styled-components';

export const StyledText = styled.span`
    color: ${props => props.theme.color[props.color] || 'inherit'};
    font-size: ${props => props.theme.scale[props.scale] || 'inherit'};
`;
