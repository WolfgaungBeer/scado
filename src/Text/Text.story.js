import React from 'react';
import { storiesOf } from '@storybook/react';
import { text } from '@storybook/addon-knobs';
import Text from './';
import Icon from '../Icon';


storiesOf('Text', module)
    .add('H1', () => (
        <Text.H1>Das ist ein Text</Text.H1>
    ))
    .add('H2', () => (
        <Text.H2>Das ist ein Text</Text.H2>
    ))
    .add('H3', () => (
        <Text.H3>Das ist ein Text</Text.H3>
    ))
    .add('H4', () => (
        <Text.H4>Das ist ein Text</Text.H4>
    ))
    .add('H5', () => (
        <Text.H5>Das ist ein Text</Text.H5>
    ))
    .add('H6', () => (
        <Text.H6>Das ist ein Text</Text.H6>
    ))
    .add('P', () => (
        <Text.P>Das ist ein Text</Text.P>
    ))
    .add('A', () => (
        <Text.A>Das ist ein Text</Text.A>
    ))
    .add('Label', () => (
        <Text.Label>Das ist ein Text</Text.Label>
    ))
    .add('with props', () => (
        <Text.H1
            color={text('color', 'success')}
            scale={text('scale', 'l')}
        >Das ist ein Text</Text.H1>
    ))
    .add('with an icon on the left side', () => (
        <div>
            <Icon icon="dashboard" color="success" scale="xl" />
            <Text.Span color="success" scale="xl">Das ist ein Text</Text.Span>
        </div>
    ))
    .add('with an icon on the right side', () => (
        <div>
            <Text.Span color="success" scale="xl">Das ist ein Text</Text.Span>
            <Icon icon="dashboard" color="success" scale="xl" />
        </div>
    ))
    .add('with an nested icon', () => (
        <Text.Span color="success" scale="xl">
            <Icon icon="dashboard" color="success" scale="xl" />
            Das ist ein Text
        </Text.Span>
    ));
