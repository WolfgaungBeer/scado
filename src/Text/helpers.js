export const textStyle = () => props => `
    color: ${props.theme.color[props.color]};
    font-size: ${props.theme.scale[props.scale]};
`;

export const font = type => props => `
    font-family: ${props.theme.fonts[type]};
`;
