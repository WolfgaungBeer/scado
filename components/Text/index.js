import React from 'react';
import { string, node } from 'prop-types';
import { StyledText } from './styled';


const propTypes = {
    color: string,
    scale: string,
    children: node,
};

const defaultProps = {
    color: undefined,
    scale: undefined,
    children: undefined,
};

const Text = ({ color, scale, children }) => (
    <StyledText color={color} scale={scale}>
        {children}
    </StyledText>
);

Text.propTypes = propTypes;
Text.defaultProps = defaultProps;

export default Text;
