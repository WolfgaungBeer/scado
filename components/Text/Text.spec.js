import React from 'react';
import renderer from 'react-test-renderer';
import Wrapper from '../../__mocks__/Wrapper';
import Text from './';

it('renders the Text Component correctly', () => {
    const tree1 = renderer.create(
        <Wrapper>
            <Text>Das ist ein Text</Text>
        </Wrapper>,
    ).toJSON();
    expect(tree1).toMatchSnapshot();

    const tree2 = renderer.create(
        <Wrapper>
            <Text color="error" scale="xl">Das ist ein Text</Text>
        </Wrapper>,
    ).toJSON();
    expect(tree2).toMatchSnapshot();
});
