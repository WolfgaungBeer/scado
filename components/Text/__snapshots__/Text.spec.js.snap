// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Text Component correctly 1`] = `
<span
  className="sc-bdVaJa iLwsQj"
  color={undefined}
  scale={undefined}
>
  Das ist ein Text
</span>
`;

exports[`renders the Text Component correctly 2`] = `
<span
  className="sc-bdVaJa hRHokw"
  color="error"
  scale="xl"
>
  Das ist ein Text
</span>
`;
